src/librustdoc/lib.rs

```diff
@@ -444,7 +444,8 @@ where R: 'static + Send,
 
     let (tx, rx) = channel();
 
-    let result = rustc_driver::report_ices_to_stderr_if_any(move || {
+    rustc_driver::install_ice_hook();
```

Comment:
This may cause problem for tests. We don't want to use the custom panic hook for them.

Reply:
I've played around with this for a bit and couldn't notice any rustdoc changes, except that ICEs while building a test are missing the query stack for some reason, which seems odd.

Reply:
Correction: The hook in this PR never seems to be called when doing `rustdoc --test bla.rs`. On nightly, the query stack hook still runs, but the normal ICE message isn't printed.

`run_compiler` is apparently only called for building documentation, not for running tests.

I'll change it to also register the hook when running doctests, since that matches the old behavior more closely, and having proper ICE messages there is a useful thing.

Reply:
The ICE message printed by the hook ends up escaping libtest's output capturing, so it looks really jarring. I've removed this line instead. The only thing lost is the query stack. A user will still see that an ICE is an ICE and can extract it into a non-doctest instead.

Let me know if this isn't good enough. Ideally we'd rework how rustdoc runs tests entirely so it behaves closer to the normal rustc frontend, but that's clearly out of scope for this PR.